handler/src/main/java/io/netty5/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -2216,7 +2224,19 @@ private boolean isDestroyed() {
     }
 
     final boolean checkSniHostnameMatch(byte[] hostname) {
-        return Java8SslUtils.checkSniHostnameMatch(matchers, hostname);
+        if (matchers != null && !matchers.isEmpty()) {
+            SNIHostName name = new SNIHostName(hostname);
+            Iterator<SNIMatcher> matcherIt = (Iterator<SNIMatcher>) matchers.iterator();
+            while (matcherIt.hasNext()) {
+                SNIMatcher matcher = matcherIt.next();
```

Comment:
We can use a `for` loop here, and don't need to cast anymore.

Reply:
fixed..